src/oracle/oracle.cairo

```diff
@@ -352,19 +700,63 @@ mod Oracle {
         /// * `token` - The token to get the price for.
         /// # Returns
         /// The signers
-        fn get_signers(
-            self: @ContractState, data_store: IDataStoreDispatcher, params: SetPricesParams,
+        fn get_signers_(
+            self: @ContractState, data_store: IDataStoreDispatcher, params: @SetPricesParams,
         ) -> Array<ContractAddress> {
-            // TODO
+            let mut signers: Array<ContractAddress> = array![];
+
+            let signers_len = *params.signer_info & bits::BITMASK_16;
+
+            if signers_len < data_store.get_u128(keys::min_oracle_signers()) {
+                OracleError::MIN_ORACLE_SIGNERS(
+                    signers_len, data_store.get_u128(keys::min_oracle_signers())
+                );
+            }
+
+            if signers_len > MAX_SIGNERS {
+                OracleError::MAX_ORACLE_SIGNERS(signers_len, MAX_SIGNERS);
+            }
+
+            let mut signers_index_mask = Mask { bits: 0 };
+
+            let mut len = signers_len;
+            loop {
+                if len.is_zero() {
+                    break;
+                }
+
+                let signer_index: u128 = BitShift::shr(
+                    *params.signer_info, (16 + 16 * len) & bits::BITMASK_16
```

Comment:
8 + 8 and use BITMASK_8 instead since we're using u128

Reply:
changed